### PR TITLE
[wip] Lmdb: AbstractKV and DB interfaces

### PR DESCRIPTION
**Pull request by AskAlexSharov**
It's based on 2017 release of LMDB: 0.9.19 (last version is 0.9.24).
Will update bindings in another PR (because I don't know amout of work there). 

- Introduced `ethdb.ObjectDatabase` - which is analog of BoltDB/BadgerDB classes but works on top of AbstractKV. I called it `Object`, because KV provides "Iterator-style API" while this class provides "Object-style API". 
- Added flag: `--database=lmdb`
- PutMulti - not supported yet, because need more investigation how it works in LMDB

### CODE REVIEW

File: trie/flatdb_sub_trie_loader.go [hunk 1]
```diff
@@ -347,6 +350,9 @@ func (fstl *FlatDbSubTrieLoader) iteration(c, ih ethdb.Cursor, first bool) error
 		if fstl.ihK, fstl.ihV, err = ih.Next(); err != nil {
 			return err
 		} // go to children, not to sibling
+		if err != nil {
```
Comment (AlexeyAkhunov) [inline trie/flatdb_sub_trie_loader.go:353]: Why?

File: trie/flatdb_sub_trie_loader.go [hunk 2]
```diff
@@ -335,6 +335,9 @@ func (fstl *FlatDbSubTrieLoader) iteration(c, ih ethdb.Cursor, first bool) error
...
```
Comment (AlexeyAkhunov) [inline trie/flatdb_sub_trie_loader.go:338]: Why?

File: trie/flatdb_sub_trie_loader.go [hunk 3]
```diff
@@ -653,6 +659,9 @@ func (fstl *FlatDbSubTrieLoader) LoadSubTries() (SubTries, error) {
 			if !bytes.Equal(k, prefix) {
 				panic(fmt.Sprintf("IH and WitnessSize buckets must have same keys set: %x, %x", k, prefix))
 			}
+			if k != nil && len(v) == 0 {
```
Comment (AlexeyAkhunov) [inline trie/flatdb_sub_trie_loader.go:662]: When does this happen?

File: trie/flatdb_sub_trie_loader.go (hunk 2 above)
Comment (AskAlexSharov) [reply trie/flatdb_sub_trie_loader.go:338]: merge conflict, removed

File: trie/flatdb_sub_trie_loader.go (hunk 3 above)
Comment (AskAlexSharov) [reply trie/flatdb_sub_trie_loader.go:662]: already fixed the case, removed.

**Pull request merged**